### PR TITLE
Changes to support BBEdit

### PR DESCRIPTION
**Pull request by tjparnell**
Per request in #86, these are the shutdown and exit server changes initially proposed by @mixio in #66 and cleaned up of extraneous code. PLS now works as expected with BBEdit version 14.1 (current latest). I have also included a short blurb in the README regarding implementation with BBEdit, plus a couple minor formatting changes.

### CODE REVIEW

**Comment by FractalBoy** [comment]
Thank you! I will try to take a look and merge this week.

File: server/lib/PLS/Server/Request/Shutdown.pm [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package PLS::Server::Request::Shutdown;
+
+use strict;
+use warnings;
+
+use parent 'PLS::Server::Request';
+
+use Scalar::Util qw(blessed);
```
[the remark this answers is not in the document]
Comment (FractalBoy) [reply server/lib/PLS/Server/Request/Shutdown.pm:8]:
```suggestion
```

File: server/lib/PLS/Server/Request/Exit.pm [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package PLS::Server::Request::Exit;
+
+use strict;
+use warnings;
+
+use parent 'PLS::Server::Request';
+
+use Scalar::Util qw(blessed);
```
[the remark this answers is not in the document]
Comment (FractalBoy) [reply server/lib/PLS/Server/Request/Exit.pm:8]:
```suggestion
```

File: server/lib/PLS/Server/Request/Shutdown.pm (hunk 1 above)
[the remark this answers is not in the document]
Comment (FractalBoy) [reply server/lib/PLS/Server/Request/Shutdown.pm:8]:
```suggestion
use PLS::Server::Response::Shutdown;
use PLS::Server::State;
```

File: server/lib/PLS/Server/Request/Shutdown.pm [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+This is a notification message from the client to the server requesting
+that the server shuts down.
+
+=cut
+
+sub service
+{
+    my ($self) = @_;
+
```
Comment (FractalBoy) [inline server/lib/PLS/Server/Request/Shutdown.pm:24]:
```suggestion

$PLS::Server::State::SHUTDOWN = 1;
```

**Comment by FractalBoy** [comment]
@tjparnell I went ahead and made some changes since there were ultimately many changes I wanted to make.

Can you test and confirm that it works with BBEdit?

**Comment by tjparnell** [comment]
Thanks @FractalBoy for incorporating the changes and making improvements. I've tested it on BBEdit and it continues to work as before. No issues.

**Pull request merged**